### PR TITLE
Group icons

### PR DESCRIPTION
**Pull request by G3rrus**
- Completely overhauled group tracking system in style as suggested by Alex2k.
- Changed some files so now the component is actually called group tracking as it should be.
- Updated the unitTracker in a way to support the new type of unit tracking planned once the repo-update with the new marker hits. (speed that updates marker pos might still need to be adjusted).

### CODE REVIEW

**Review by Alextwokay — APPROVED** [review]
Seems to work nicely.
A good feature just got upgraded to great!

**Review by Liquidtracer — APPROVED** [review]
Like it, worth waiting for repo update before merging?

File: framework/player/init/groupTracker.sqf [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+	clearGroupIcons _grp;
+	_grp addGroupIcon [_type,[0,0]];
+	_grp setgroupIconParams [var_iconColor,_txt,_scale,true];
+};
+
+
+// WHICH GROUPS GET TRACKED ///////////////////////////////////////////////////////////////////////
+//FOX PLATOON
+if !(isNil "Grp_FOX_6") then {[Grp_FOX_6,"b_hq","FOX6",1] call _lmf_fnc_createIcon};
```
Comment (diwako) [inline framework/player/init/groupTracker.sqf]: You can remove the 1 as last parameter, the private function already does that, unless you want some other size, what is the point.

File: framework/player/init/groupTracker.sqf [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+	if (_mouseButton isEqualTo 0) then {
+		//GET ARRAY WITH UNIT NAMES FROM GROUP
+		private _nameArray = [];
+		{
+			private _name = name _x;
+			_nameArray pushBackUnique _name;
+		} forEach units _group;
+
+		private _nameList = _nameArray joinString "<br/>";
+		private _br = format ["<br/>"];
```
Comment (diwako) [inline framework/player/init/groupTracker.sqf]: What is the point of `_br`? this so far seems useless as you format a text that does not need formatting.

File: framework/player/init/groupTracker.sqf [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+		private _groupID = format ["<t shadow='2' color='#FFBA26' size='0.75'align='center'>%1</t>",groupID _group];
+		private _names = format ["<t shadow='2' color='#D7DBD5' size='0.5'align='center'>%1</t>",_nameList];
+
+		//CREATE LAYER AND SPAWN TEXT
+		private _layerID = ["lmf_layer2"] call BIS_fnc_rscLayer;
+		[_groupID + _br + _names,(getMousePosition#0) - 0.65,(getMousePosition#1) - 0.1,10,0,0,_layerID] spawn BIS_fnc_dynamicText;
```
Comment (diwako) [inline framework/player/init/groupTracker.sqf]: use format here, like that is the prime use for it, concatenating strings or inserting into strings.

File: framework/player/init/groupTracker.sqf [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+private _clickIconEH = addMissionEventHandler ["GroupIconClick",{
+	params ["_is3D","_group","_waypointId","_mouseButton"];
+
+	//CHECK FOR LEFT CLICK
+	if (_mouseButton isEqualTo 0) then {
+		//GET ARRAY WITH UNIT NAMES FROM GROUP
+		private _nameArray = [];
+		{
+			private _name = name _x;
+			_nameArray pushBackUnique _name;
```
Comment (diwako) [inline framework/player/init/groupTracker.sqf]: Doubt we have two John Does, so `pushbackunique` is probably not needed. Also no need for assigning a variable if you do not use the name a second time.

File: framework/player/init/unitTracker.sqf [hunk 5]
```diff
@@ -12,35 +13,39 @@ lmf_player_tracklist = [];
...
-		_marker setmarkerSizeLocal [0.4,0.4];
+	private _fnc_addUnit =  {
+		private _unit = _this#0;
 		lmf_player_tracklist pushbackunique _unit;
 	};
+
 	while {true} do {
+		//ADD PLAYERS FROM PLAYER GROUP TO LIST (MINUS GROUP LEADER)
+		{
+			if (_x in lmf_player_tracklist) then {} else {0 =[_x] call _fnc_addUnit};
```
Comment (diwako) [inline framework/player/init/unitTracker.sqf]: findif and negate the condition so the function call is in the `then` block instead. Also `0= ... call ... ` is crap as your do not use your if statement to assign a variable.

File: framework/player/init/unitTracker.sqf [hunk 6]
```diff
@@ -12,35 +13,39 @@ lmf_player_tracklist = [];
...
-	waitUntil {count lmf_player_tracklist > 0};
+private _trackingEH = ((findDisplay 12) displayCtrl 51) ctrlAddEventHandler ["Draw",{
+	if (!visibleMap || {lmf_player_tracklist isEqualTo []}) exitWith {};
 	{
-		private _marker = ("nk_m_"+ str(_x));
-		_marker setmarkerposlocal getposASL _x;
-		if (isNull _x) then {
-			lmf_player_tracklist = lmf_player_tracklist - [_x];
-			deleteMarkerLocal _marker;
+		private _color = [assignedTeam _x] call {
```
Comment (diwako) [inline framework/player/init/unitTracker.sqf]: Pass `assignedTeam _x # 0` into the function, so you do not need a `select` inside it.

File: framework/player/init/unitTracker.sqf [hunk 7]
```diff
@@ -12,35 +13,39 @@ lmf_player_tracklist = [];
...
+			if (_this#0 == "RED") exitwith {[1,0,0,1]};
+			if (_this#0 == "GREEN") exitwith {[0,1,0,1]};
+			if (_this#0 == "BLUE") exitwith {[0,0,1,1]};
+			if (_this#0 == "YELLOW") exitwith {[1,1,0,1]};
+			[1,1,1,1]
 		};
+		(_this#0) drawIcon ["iconMan",[0,0,0,1],getPosASL _x,20,20,getDir _x];
```
Comment (diwako) [inline framework/player/init/unitTracker.sqf]: If you need to use a `select` statement more then once, might consider of loading what you need into a private variable. In this case you are doing `_this#0`for ever unit tracked, while you only need to assign it once outside the `count`

File: settings/cfg_Mission.sqf [hunk 8]
```diff
@@ -50,5 +50,5 @@ var_keepRole = true; // Should players keep their role upon respawn? (default: t
 
 // MARKERS ////////////////////////////////////////////////////////////////////////////////////////
 var_markerSide = "ColorWEST"; // What colorsheme should markers on playerside use? ("ColorWEST", "ColorEAST", "ColorGUER") (default: "ColorWEST")
-var_groupMarkers = true; // Use group-icons on map. (default: true)
+var_groupTracker = true; // Use group-icons on map. (default: true)
```
Comment (diwako) [inline settings/cfg_Mission.sqf:53]: I hope no one is copy pasting their `cfg_Mission.sqf` from another mission then ;)

**Review by diwako — CHANGES_REQUESTED** [review]

File: settings/cfg_Mission.sqf (hunk 8 above)
Comment (G3rrus) [reply settings/cfg_Mission.sqf:53]: > 
> 
> I hope no one is copy pasting their `cfg_Mission.sqf` from another mission then ;)

I don't think that will be an issue at this point (hopefully)

**Review by diwako — APPROVED** [review]

**Pull request merged**